clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5035,16 +5035,14 @@ static void ProcessVSRuntimeLibrary(const ArgList &Args,
         !Args.hasArg(options::OPT_nolibsycl)) {
       if (RTOptionID == options::OPT__SLASH_MDd) {
         if (Args.hasArg(options::OPT_fpreview_breaking_changes))
-          CmdArgs.push_back("--dependent-lib=sycl" SYCL_MAJOR_VERSION
-                            "-previewd");
+          CmdArgs.push_back("--dependent-lib=sycl-previewd");
         else
-          CmdArgs.push_back("--dependent-lib=sycl" SYCL_MAJOR_VERSION "d");
+          CmdArgs.push_back("--dependent-lib=sycld");
```

Comment:
As the versions progress - is there any way of verifying that the proper version of the library is being picked up in case there is some overlap in environment when resolving the libraries?

Reply:
I was sure that it is guaranteed that we take correct .lib file from "lib" directory of compiler installation because clang driver adds /libpath option with full path like this:
/libpath:[path to lib directory] /defaultlib:sycl.lib

From the description of the libpath option:
https://learn.microsoft.com/en-us/cpp/build/reference/libpath-additional-libpath?view=msvc-170
"Use the /LIBPATH option to override the environment library path. The linker will first search in the path specified by this option, and then search in the path specified in the LIB environment variable."

It indeed works as described. I.e. even if there are multiple installations of different sycl compilers and LIB env variable is set to point to all of them, we are going to link with proper .lib file even if it doesn't have version in the name. More over, for example, we link with msvcrt lib which doesn't have version as well.

So, I was trying to create additional test for this by your request.
But while experimenting I realized that there is a special case: current directory.
 For example, if there is a file named sycl.lib (or sycl7.lib) in the current directory, then it is used at linkage regardless of libpath option or LIB env variable.
It is possible to avoid that issue if to use full path to .lib libraries like this: https://github.com/intel/llvm/commit/862994cc8ee16ea830293dd88662b618b429ecb2

I don't know and can't explain why msvc linker behaves that way for current directory and if this case is worth being worried about.

Since original intent of these changes wasn't to get rid of version from clang driver, I am going to revert that part of the changes in this PR. I can add them in further PR if we consider worth it.

Reply:
Thanks @againull - that sounds like a good plan.